### ISSUE TITLE
Replace contact, location and social media fields on landing page with contact paragraph

### ISSUE THREAD

**Issue by stephen-cox** [localgovdrupal/localgov_services#46]
As per the user story for the service landing page (https://trello.com/c/kH5UvU3P/203-as-a-content-designer-create-and-manage-services-landing-page) we remove all the contact, location and social media fields and use a contact paragraph in their place.

**Comment by tanc** [comment]
Yes I think this makes the most sense as the field approach is less flexible and requires more markup in the node template when it would be better providing some default markup for the paragraph template and keeping the node template mostly default. Even deleting the currently provided `node--localgov-services-landing--full` template and letting the config/view mode dictate the position of the fields

**Comment by finnlewis** [comment]
Yep - agreed. Allowing more flexibility in the view mode field configuration makes total sense, so flipping this group of fields into a paragraph will help with this. It also helps with styling / theming in a more component based approach. 

Just pinging @andybroomfield and @Adnan-cds in case you guys have any strong opinions....

**Comment by finnlewis** [comment]
And @cjstevens78, sorry, do you have any thoughts on this?

**Comment by stephen-cox** [comment]
If we're going to do this we want to do this sooner rather than later. Basically before we reach a stage when we have to automate a migration of data out of node fields in into a paragraph.

**Comment by andybroomfield** [comment]
No objections, though will need to amend the migration for services provided by localgov_legacy_migration. Should also help in some way resolve #75 and https://github.com/localgovdrupal/localgov_core/issues/35.

**Comment by ekes** [comment]
So @tanc has done this for Lambeth already.

This is an 'API' changing modification. So we should work out how to implement it in localgov drupal generally.
We can write a update hook, or script that can be optionally run, to move any data in the service landing pages into contact paragraphs, remove the fields, and enable the contact paragraph field.
Doing this in a update hook automatically would be obvious. And there aren't people dependent on the fields yet? (Actual question). But if there were implementations with themes that assume the fields it would break peoples sites (hence calling it an API changing modification).

This is where - and I'm just making this up as I think - it could be a script:
 * The old contact fields are removed from install in the module, so new sites don't get them, but old sites they are in their configuration already;
 * The new field and related paragraphs are added. New sites get it on the services landing bundle and enabled. An update hook also adds it to old sites - but not visible on the form or display modes.
 * There is an upgrade script to run with drush that administrators can run that copies the data into the new paragraphs, attaches the new paragraphs to the correct services, deletes the old fields, enables the new one on the form and display modes.
 * There is information about the script and the change in the release notes of the new minor (?) release.

As a pattern this would enable administrators to upgrade the site generally; and only move the field once tested in dev/staging and their custom theme has had any required alterations.

**Comment by ekes** [comment]
> As a pattern this would enable administrators to upgrade the site generally; and only move the field once tested in dev/staging and their custom theme has had any required alterations.

I'm now thinking a drush command that can list all the pending 'user controlled' updates with descriptions.
So I'm now thinking, I've either missed something that makes this impossible, or someone's done this already somewhere?

**Comment by stephen-cox** [comment]
The only people I think are likely to be using current contact fields will be Lambeth. It should be easy enough to check if they are actually being used or not.

Alhtough we do need to decide how best to handle changes like this. The Drush script route implies that some sites will adopt the new fields and some won't. This means we will be supporting the old fields and the new ones, which will be extra work to maintain, but does seem less risky for sites adopting the changes as it pushes some of the responsibility for ensuring everything updates okay. Using update hooks insures all sites have the same base configuration, should be easier for us to maintain, but should all sites using the distribution be forced to adopt all the changes?

My gut feeling is that if sites can choose what they adopt we may end up with serious complications later on when API change rely on other API changes.

> I'm now thinking a drush command that can list all the pending 'user controlled' updates with descriptions.
So I'm now thinking, I've either missed something that makes this impossible, or someone's done this already somewhere?

It would certainly be useful if there's a standard way of dealing with such changes using Drush.

**Comment by andybroomfield** [comment]
For now this should be resolvable with a migration. Ideally change everyone to the paragraphs by default.

However, this would be a good time to test whether its possible to upgrade node config whilst maintiaining councils own customisations. Some of our service page nodes will have custom fields that are very BHCC specific in the near future, and we need a way to upgrade the node config but leaving our own changes intact.

**Comment by ekes** [comment]
> This means we will be supporting the old fields and the new ones,

We remove them from the distro; and 

>> I'm now thinking a drush command that can list all the pending 'user controlled' updates with descriptions.

With documentation and/or a list in drush explain that they are no longer supported and it's the responsibility of the site maintainer to update.

> this would be a good time to test whether its possible to upgrade node config whilst maintiaining councils own customisations
> Some of our service page nodes will have custom fields that are very BHCC specific

And in all we do we shouldn't be breaking _additional_ fields that sites have added. Advice - again we should document this - would be that sites prefix all their custom fields with `sitespecificprefix_` just as we do `localgov_` to keep things even less likely to clash.

**Comment by andybroomfield** [comment]
This came up on another ticket
https://github.com/localgovdrupal/localgov/issues/213

Is porting still going ahead?

**Comment by keelanfh** [comment]
On a fresh install, the contact fields are still appearing. Even if we're not able to migrate for existing users of these fields, it would seem sensible to me to move away from showing these options in the default config. We got quite confused about this when getting started, and then hit a barrier when we couldn't use these fields on other content types. The contact component is much more flexible.

**Comment by andybroomfield** [comment]
The reference issue was not a PR, but duplicate of this issue.

We should check how many councils actually use the fields vs the contact cards in page components. From BHCC side we've never used page components on services pages, but are starting to use them in paragraph layouts (see below).
On the few service landing pages that do require them, we have used the fields, though a future paragraph will replace there use.

Maybe worth putting a separate issue about if page components are way forward and if widely used, as many councils have replaced the body field with layout paragraphs and presumably get the paragraphs like the contact card that way.
Page components are useful as they are rebadged version of paragraphs library, and the from library paragraph and the ability to promote a paragraph to library directly from a page is quite useful.
